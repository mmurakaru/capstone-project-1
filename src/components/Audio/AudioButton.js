import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as PlayIcon } from '../../assets/play.svg'
import { ReactComponent as PauseIcon } from '../../assets/pause.svg'
import useAudio from './useAudio'

AudioButton.propTypes = {
  songUrl: PropTypes.string,
  id: PropTypes.string,
  currentSong: PropTypes.string,
  setCurrentSong: PropTypes.func,
}

export default function AudioButton({
  songUrl,
  id,
  currentSong,
  setCurrentSong,
}) {
  const { isSongPlaying, toggleAudioState } = useAudio({
    songUrl,
    id,
    currentSong,
  })

  return (
    <Button onClick={handleAudio}>
      {isSongPlaying ? <PauseIcon title="pause" /> : <PlayIcon title="play" />}
    </Button>
  )

  function handleAudio() {
    setCurrentSong(id)
    toggleAudioState()
  }
}

const Button = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  fill: var(--primary-dark);
  height: 46px;
  padding: 10px;
  width: 46px;
`
