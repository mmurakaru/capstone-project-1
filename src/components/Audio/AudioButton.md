Audio Button

```js
<AudioButton
  songUrl={
    'https://p.scdn.co/mp3-preview/3011a4d543fca575e3f89bdb80f80c559161a96c?cid=5f7fa324ec124c5da5672dd5d6fa9b2f'
  }
  id={'nj8se0eqez237'}
  currentSong={'nj8se0eqez237'}
  setCurrentSong={() => {}}
  handleAudio={() => {}}
/>
```
