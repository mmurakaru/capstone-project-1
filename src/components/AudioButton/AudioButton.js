import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as PlayIcon } from '../../assets/play.svg'
import { ReactComponent as PauseIcon } from '../../assets/pause.svg'
import useAudio from './useAudio'

AudioButton.propTypes = {
  songUrl: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  currentSong: PropTypes.string.isRequired,
  setCurrentSong: PropTypes.func.isRequired,
}

export default function AudioButton({
  songUrl,
  id,
  currentSong,
  setCurrentSong,
}) {
  const { isSongPlaying, setIsSongPlaying } = useAudio({
    songUrl,
    id,
    currentSong,
  })

  return (
    <Button onClick={handleAudio}>
      {isSongPlaying ? <PauseIcon /> : <PlayIcon />}
    </Button>
  )

  function handleAudio() {
    setCurrentSong(id)
    setIsSongPlaying(!isSongPlaying)
  }
}

const Button = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  fill: var(--primary-dark);
  height: 46px;
  padding: 10px;
  width: 46px;
`
