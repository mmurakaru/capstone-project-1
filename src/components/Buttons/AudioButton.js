import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as PlayIcon } from '../../assets/play.svg'
import { ReactComponent as PauseIcon } from '../../assets/pause.svg'

AudioButton.propTypes = {
  handleClick: PropTypes.func,
  isSongPlaying: PropTypes.bool.isRequired,
}

export default function AudioButton({ handleClick, isSongPlaying }) {
  return (
    <Button onClick={handleClick}>
      {isSongPlaying ? <PauseIcon title="pause" /> : <PlayIcon title="play" />}
    </Button>
  )
}

const Button = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  fill: var(--primary-dark);
  height: 46px;
  padding: 10px;
  width: 46px;
`
