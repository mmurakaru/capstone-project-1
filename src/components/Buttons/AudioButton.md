Audio Button: Play

```js
<AudioButton handleClick={() => {}} isSongPlaying={false} />
```

Audio Button: Pause

```js
<AudioButton handleClick={() => {}} isSongPlaying={true} />
```
