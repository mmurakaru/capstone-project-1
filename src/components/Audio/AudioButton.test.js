import AudioButton from './AudioButton'
import { render } from '@testing-library/react'
import user from '@testing-library/user-event'

describe('AudioButton', () => {
  it('renders correctly', () => {
    const { container } = render(
      <AudioButton
        songUrl={'https://www.testurl.com'}
        id={'nj8se0eqez237'}
        currentSong={'nj8se0eqez237'}
        setCurrentSong={() => {}}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows play button per default', () => {
    const { isSongPlaying, getByTitle } = render(
      <AudioButton
        songUrl={'https://www.testurl.com'}
        id={'nj8se0eqez237'}
        currentSong={'nj8se0eqez237'}
        setCurrentSong={() => {}}
      />
    )
    expect(isSongPlaying).toBeFalsy()
    expect(getByTitle('play')).toBeInTheDocument()
  })

  it('changes the state isSongPlaying by click', () => {
    const { getByRole, queryByTitle, getByTitle } = render(
      <AudioButton
        songUrl={'https://www.testurl.com'}
        id={'nj8se0eqez237'}
        currentSong={'nj8se0eqez237'}
        setCurrentSong={() => {}}
      />
    )
    expect(getByTitle('play')).toBeInTheDocument()

    const button = getByRole('button')
    user.click(button)

    expect(getByTitle('pause')).toBeInTheDocument()
    expect(queryByTitle('play')).not.toBeInTheDocument()
  })
})
