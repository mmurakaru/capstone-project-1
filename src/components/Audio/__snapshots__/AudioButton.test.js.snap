// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioButton renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  fill: var(--primary-dark);
  height: 46px;
  padding: 10px;
  width: 46px;
}

<button
  class="c0"
>
  <svg
    title="play"
  >
    play.svg
  </svg>
</button>
`;
